task5/src/main/test/com/clouway/adapter/http/BankOperationHandlerTest.java

```diff
@@ -0,0 +1,86 @@
+package com.clouway.adapter.http;
+
+import com.clouway.core.UserRepository;
+import com.clouway.core.Validator;
+import org.jmock.Expectations;
+import org.jmock.auto.Mock;
+import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 08.06.16.
+ */
+public class BankOperationHandlerTest {
+  private BankOperationHandler bankOperationHandler;
+
+  @Rule
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Mock
+  Validator validator;
+
+  @Mock
+  UserRepository userRepository;
+
+  @Before
+  public void setUp() {
+    bankOperationHandler = new BankOperationHandler(userRepository, validator);
+  }
+
+  @Test
+  public void makeDepositHappyPath() throws Exception {
+    final String amount = "200.00";
+    final String email = "admin@abv.bg";
+    final Double amountInDouble = 200.00D;
+
+    context.checking(new Expectations() {{
+      oneOf(request).getParameter("operation");
+      will(returnValue("Deposit"));
+      oneOf(request).getParameter("email");
+      will(returnValue(email));
+      oneOf(request).getParameter("amount");
+      will(returnValue(amount));
+
+      oneOf(validator).isValid(amount);
+      will(returnValue(true));
+
+      oneOf(userRepository).makeDeposit(amountInDouble, email);
+      oneOf(response).sendRedirect("/useraccount?email=" + email);
+    }});
+    bankOperationHandler.doPost(request, response);
+  }
+
+  @Test
+  public void makeWithdrawHappyPath() throws Exception {
+    final String amount = "100.00";
+    final String email = "admin@abv.bg";
+    final Double amountInDouble = 100.00D;
+
+    context.checking(new Expectations() {{
+      oneOf(request).getParameter("operation");
+      will(returnValue("Withdraw"));
+      oneOf(request).getParameter("email");
+      will(returnValue(email));
+      oneOf(request).getParameter("amount");
+      will(returnValue(amount));
+
+      oneOf(validator).isValid(amount);
+      will(returnValue(true));
+
+      oneOf(userRepository).makeWithdraw(amountInDouble, email);
+      oneOf(response).sendRedirect("/useraccount?email=" + email);
+    }});
+    bankOperationHandler.doPost(request, response);
+  }
+}
```

Comment:
Not all scenarios are covered.
- invalid amounts
- withraw with not enough money